### ISSUE TITLE
Supervised Finetuning - How to mask loss for prompts 

### ISSUE THREAD

**Issue by jvhoffbauer** [huggingface/trl#542]
How can I mask the loss in supervised fine-tuning for prompts similar to how it is done in the LLAMA-2 paper? 

Specifically, I have a dataset of prompts and ideal answers. When fine-tuning my model with a `SFTTrainer` using a `ConstantLengthDataset` (similar to the StackExchange example), how can I ensure that prompts are not considered in the loss?

**Comment by BramVanroy** [comment]
This PR is related https://github.com/lvwerra/trl/pull/456

**Comment by jvhoffbauer** [comment]
@BramVanroy If I understand correctly, the `DataCollatorForCompletionOnlyLM` already partially solves this issue. However, when using a ConstantLengthDataset, it would detect only the first `### Response:` and treat consecutive `### Prompt: ... ### Response` as part of the first response.

**Comment by BramVanroy** [comment]
Indeed, there's also a PR for ChatCompletion masking as you have found. Linking here for completeness' sake. https://github.com/lvwerra/trl/pull/456

**Comment by younesbelkada** [comment]
Hi @jvhoffbauer 
Yes, I second what @BramVanroy said, you might be interested to use `DataCollatorForCompletionOnlyLM` , feel free to directly chime in #456

**Comment by jvhoffbauer** [comment]
Yes, happy to help if it makes sense!

**Comment by younesbelkada** [comment]
FYI, here is the section on the docs on how to FT on completions only using `DataCollatorForCompletionOnlyLM`: https://huggingface.co/docs/trl/main/en/sft_trainer#train-on-completions-only

**Comment by jvhoffbauer** [comment]
Solved by https://github.com/huggingface/trl/pull/456
